lib/std/jsfetch.nim

```diff
@@ -0,0 +1,139 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+when not defined(js):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+
+import asyncjs, jsheaders
+from httpcore import HttpMethod
+
+type
+  FetchOptions* = ref object  ## Options for Fetch API.
+    keepalive*: bool
+    metod* {.importjs: "method".}: cstring
+    body*, integrity*, referrer*, mode*, credentials*, cache*, redirect*, referrerPolicy*: cstring
+
+  FetchModes* = enum  ## JavaScript Fetch API mode options.
+    fmCors = "cors".cstring
```

Comment:
ask @Araq whether this is ok; usually `$x` returns a string, but here it'd return a cstring

Reply:
It's not ok at all.

Reply:
```nim
type FetchModes = enum
  fmCors = "cors".cstring
  fmNoCors = "no-cors".cstring
  fmSameOrigin = "same-origin".cstring

doAssert $fmCors is string
```

:shrug:

Reply:
What's wrong with:

```nim

type FetchModes = enum
  fmCors = "cors"
  fmNoCors = "no-cors"
  fmSameOrigin = "same-origin"

```